backend/dal/PIMSContextFactory.cs

```diff
@@ -45,27 +45,44 @@ public PimsContextFactory()
         public PimsContext CreateDbContext(string[] args)
         {
             DotNetEnv.Env.Load();
-            string environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
+            string environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Production";
 
-            var config = new ConfigurationBuilder()
+            // As per Microsoft documentation, a typical sequence of configuration providers is:
+            //   1. appsettings.json
+            //   2. appsettings.{Environment}.json
+            //   3. Secret Manager (if in development)
+            //   4. Environment variables using the Environment Variables configuration provider.
+            //   5. Command - line arguments using the Command-line configuration provider.
+            // source: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#configuration-providers
+            var builder = new ConfigurationBuilder()
                 .SetBasePath(Directory.GetCurrentDirectory())
-                .AddEnvironmentVariables()
-                .AddJsonFile("connectionstrings.json", optional : true, reloadOnChange : true)
-                .AddJsonFile($"connectionstrings.{environment}.json", optional : true, reloadOnChange : true)
-                .AddUserSecrets<PimsContext>()
-                .Build();
+                .AddJsonFile("connectionstrings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"connectionstrings.{environment}.json", optional: true, reloadOnChange: true);
+
+            if (!this.isProduction(environment))
+            {
+                builder.AddUserSecrets<PimsContext>();
+            }
+
+            builder.AddEnvironmentVariables();
+
+            var config = builder.Build();
 
             var cs = config.GetConnectionString("PIMS");
-            var builder = new SqlConnectionStringBuilder(cs)
+            var sqlBuilder = new SqlConnectionStringBuilder(cs)
             {
                 Password = config["DB_PASSWORD"]
             };
 
             var optionsBuilder = new DbContextOptionsBuilder<PimsContext>();
-            // optionsBuilder.UseNpgsql (cs, opts => opts.CommandTimeout ((int) TimeSpan.FromMinutes (10).TotalSeconds));
-            optionsBuilder.UseSqlServer(builder.ConnectionString, opts => opts.CommandTimeout((int) TimeSpan.FromMinutes(10).TotalSeconds));
+            optionsBuilder.UseSqlServer(sqlBuilder.ConnectionString, opts => opts.CommandTimeout((int)TimeSpan.FromMinutes(10).TotalSeconds));
             return new PimsContext(optionsBuilder.Options);
         }
+
+        private bool isProduction(string env)
```

Comment:
Look for the existing extension method rather than creating a custom one.  I believe there is one already.

Reply:
I did look up the extension method you mention but this Factory class is not part of a web application - instead is being called by the `dotnet ef` cli. 

I tried to find a way to get a reference to the `IWebHostEnvironment` env which is the one that supports the extension method, and could not find a way to inject it into the class given that there's no host created... I could be wrong here, if you can find a way to inject the `env` class then we can drop the custom method. Please let me know if you know a way.

Reply:
I looks like the extension method only exists for `HostingEnvironment`, which would require additional effort to setup and configure.  I'd recommend moving your function to an extension method in our `Pims.Core` library.  Also rename with PascalCase `IsProduction()`.

Reply:
I've made the changes, thanks for your suggestions!